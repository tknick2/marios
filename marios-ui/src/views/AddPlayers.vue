<script setup lang="ts"></script>

<template>
  <h1>Add Players</h1>
  <input type="number" value="8" />
</template>
