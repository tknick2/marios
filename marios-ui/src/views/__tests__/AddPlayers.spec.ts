import { beforeEach, describe, expect, it } from 'vitest'
import { VueWrapper, mount } from '@vue/test-utils'
import AddPlayers from '../AddPlayers.vue'

describe('Add Players', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = mount(AddPlayers)
  })

  it('has an input field for the number of players with a default value of 8', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has "player name" input fields equal to the number of players', () => {
    const numberOfPlayersInput = wrapper.find('[data-test="number-of-players"]')
    const playerNameInputs = wrapper.findAll('[data-test="input-player-name"]')

    expect((numberOfPlayersInput.element as HTMLInputElement).value).toBe(8)
    expect(playerNameInputs.length).toBe(8)
  })

  it.todo('shows the "P{n}" for each player input', () => {})
})
