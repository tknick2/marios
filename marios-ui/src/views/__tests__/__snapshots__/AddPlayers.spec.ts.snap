// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add Players > has an input field for the number of players with a default value of 8 1`] = `
"<h1>Add Players</h1>
<input type=\\"number\\">"
`;
